src/lj_clib.c

```diff
@@ -120,12 +120,14 @@ static void *clib_loadlib(lua_State *L, const char *name, int global)
 		   RTLD_LAZY | (global?RTLD_GLOBAL:RTLD_LOCAL));
   if (!h) {
     const char *e, *err = dlerror();
-    if (*err == '/' && (e = strchr(err, ':')) &&
+    if (err && *err == '/' && (e = strchr(err, ':')) &&
```

Comment:
Can you add a comment with the note to explain why this `err` null check is here? Some sort of (you can rephraze it as you want):
```cpp
/* Per the standard definition of `dlerror`, this should always be safe, but we have 
 * observed dlerror() returning NULL on Android (Oculus Quest) anyway, which causes a crash. */
```
Also, optionally if possible, can you specify the version(s) of Android on which you have caught out for a crash?